File: 프로그래머스/lv2/12945. 피보나치 수/피보나치 수.kt
class Solution {
    fun solution(n: Int): Int {
        var answer = 0
        var p = arrayOfNulls<Int>(n+1)
        p[0] = 0
        p[1] = 1
        for(i in 2..n){
            p[i] = (p[i-1]!!.plus(p[i-2]!!)) % 1234567
        }
        
        return p[n]!!.toInt()
    }
}

File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.js
function solution(n) {
    let a = 0; // dp[0]
    let b = 1; // dp[1]

    for (let i = 2; i <= n; i++) {
        const temp = (a + b) % 1234567;
        a = b;
        b = temp;
    }

    return n === 0 ? 0 : b;
}
